hazelcast/src/main/java/com/hazelcast/query/impl/AndResultSet.java

```diff
@@ -28,37 +28,26 @@
 import java.util.Set;
 
 import static com.hazelcast.util.Preconditions.isNotNull;
+
 /**
  * And Result set for Predicates.
  */
 public class AndResultSet extends AbstractSet<QueryableEntry> {
     private final Set<QueryableEntry> setSmallest;
     private final List<Set<QueryableEntry>> otherIndexedResults;
     private final List<Predicate> lsNoIndexPredicates;
+    private int cachedSize;
 
     public AndResultSet(Set<QueryableEntry> setSmallest, List<Set<QueryableEntry>> otherIndexedResults,
                         List<Predicate> lsNoIndexPredicates) {
         this.setSmallest = isNotNull(setSmallest, "setSmallest");
         this.otherIndexedResults = otherIndexedResults;
         this.lsNoIndexPredicates = lsNoIndexPredicates;
+        this.cachedSize = -1;
     }
 
     public byte[] toByteArray(ObjectDataOutput out) throws IOException {
```

Comment:
I reckon it was originally meant to be used for serialization? I can see the method was removed from `OrResultSet` [quite some time ago](https://github.com/jerrinot/hazelcast/commit/ee3455b2c8c256df86b2917ef8e7af0c6292df5b#diff-773b532d77d82620bf8102c731d1f76bL31).

I would just remove it from `AndResultSet` (In a PR into master only)